runtime/src/iree/hal/allocator_heap.c

```diff
@@ -100,6 +100,39 @@ static void iree_hal_heap_allocator_query_statistics(
   });
 }
 
+static iree_status_t iree_hal_heap_allocator_query_memory_heaps(
+    iree_hal_allocator_t* IREE_RESTRICT base_allocator,
+    iree_host_size_t capacity,
+    iree_hal_allocator_memory_heap_t* IREE_RESTRICT heaps,
+    iree_host_size_t* IREE_RESTRICT out_count) {
+  const iree_host_size_t count = 1;
+  if (out_count) *out_count = count;
+  if (capacity < count) {
+    return iree_status_from_code(IREE_STATUS_OUT_OF_RANGE);
+  }
+  heaps[0] = (iree_hal_allocator_memory_heap_t){
+      .type = IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL |
+              IREE_HAL_MEMORY_TYPE_HOST_VISIBLE |
+              IREE_HAL_MEMORY_TYPE_HOST_COHERENT,
+      .allowed_usage = IREE_HAL_BUFFER_USAGE_TRANSFER |
+                       IREE_HAL_BUFFER_USAGE_DISPATCH_INDIRECT_PARAMS |
+                       IREE_HAL_BUFFER_USAGE_DISPATCH_UNIFORM_READ |
+                       IREE_HAL_BUFFER_USAGE_DISPATCH_STORAGE |
+                       IREE_HAL_BUFFER_USAGE_SHARING_EXPORT |
+                       IREE_HAL_BUFFER_USAGE_SHARING_REPLICATE |
+                       IREE_HAL_BUFFER_USAGE_SHARING_CONCURRENT |
+                       IREE_HAL_BUFFER_USAGE_SHARING_IMMUTABLE |
+                       IREE_HAL_BUFFER_USAGE_MAPPING_SCOPED |
+                       IREE_HAL_BUFFER_USAGE_MAPPING_PERSISTENT |
+                       IREE_HAL_BUFFER_USAGE_MAPPING_OPTIONAL |
+                       IREE_HAL_BUFFER_USAGE_MAPPING_ACCESS_RANDOM |
+                       IREE_HAL_BUFFER_USAGE_MAPPING_ACCESS_SEQUENTIAL_WRITE,
```

Comment:
It's not quite all usage, though it may be ok to set all bits as this is an allowed_usage check instead of required_usage - will give it a shot

Reply:
simplified the DISPATCH_* flags but the others are difficult as they can be mutually exclusive and outside of this one allocator don't really have any meaningful default. since it's a public API I also don't want to mislead users - a roll-up of SHARING for example would be tempting but it doesn't make sense for anyone to actually use.